ts/src/header-validator/source.test.ts

```diff
@@ -1285,6 +1285,21 @@ const testCases: TestCase[] = [
       },
     ],
   },
+  {
+    name: 'destination-limit-lifo-non-default-priority',
+    json: `{
+      "destination": "https://a.test",
+      "destination_limit": {
+	"priority": "123"
+      }
+    }`,
+    expectedWarnings: [
+      {
+        path: ['destination_limit', 'priority'],
+        msg: 'non-default value (123) for algorithm lifo',
```

Comment:
To make sure I'm understanding this: A source might still want to specify a non-0 priority even if it's using `lifo` because a future source may use `priority-fifo` and then the first source's priority would be considered? If so, we may want to say something to that effect in this message and make this a note instead of a warning. Maybe something like:

```
... (value may still be used by future priority-fifo source)
```
?

Reply:
yeah, that's correct, updated as suggested, thanks.

Reply:
I still think it should be a note, not a warning, as it's not really something that the developer should take action on. And let's surround the `may still be used...` part with parentheses.